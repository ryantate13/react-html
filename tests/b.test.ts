import {b} from '../src';

describe('b', () => {
    it('renders with children', () => {
        expect(b({}, 'TEST')).toMatchSnapshot();
    });
    
    it('renders arrays of children', () => {
        expect(b({}, Array(10).fill('TEST'))).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(b({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(b()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(b({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(b({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(b({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(b({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(b({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(b({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(b({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(b({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(b({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(b({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(b({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(b({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(b({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(b({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(b({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden=true', () => {
        expect(b({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden=false', () => {
        expect(b({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(b({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(b({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(b({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(b({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(b({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(b({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(b({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(b({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(b({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(b({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(b({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(b({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(b({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope=true', () => {
        expect(b({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope=false', () => {
        expect(b({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(b({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(b({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(b({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(b({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(b({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(b({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style={"color":"#000000"}', () => {
        expect(b({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex=1', () => {
        expect(b({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(b({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(b({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(b({translate: "no"})).toMatchSnapshot();
    });
});