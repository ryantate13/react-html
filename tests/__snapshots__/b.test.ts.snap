// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`b renders arrays of children 1`] = `
<b>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</b>
`;

exports[`b renders with accessKey="a,b,c,d" 1`] = `
<b
  accessKey="a,b,c,d"
/>
`;

exports[`b renders with autoCapitalize="characters" 1`] = `
<b
  autoCapitalize="characters"
/>
`;

exports[`b renders with autoCapitalize="none" 1`] = `
<b
  autoCapitalize="none"
/>
`;

exports[`b renders with autoCapitalize="off" 1`] = `
<b
  autoCapitalize="off"
/>
`;

exports[`b renders with autoCapitalize="on" 1`] = `
<b
  autoCapitalize="on"
/>
`;

exports[`b renders with autoCapitalize="sentences" 1`] = `
<b
  autoCapitalize="sentences"
/>
`;

exports[`b renders with autoCapitalize="words" 1`] = `
<b
  autoCapitalize="words"
/>
`;

exports[`b renders with children 1`] = `
<b>
  TEST
</b>
`;

exports[`b renders with className="foo bar baz" 1`] = `
<b
  className="foo bar baz"
/>
`;

exports[`b renders with contentEditable="false" 1`] = `
<b
  contentEditable="false"
/>
`;

exports[`b renders with contentEditable="true" 1`] = `
<b
  contentEditable="true"
/>
`;

exports[`b renders with dir="auto" 1`] = `
<b
  dir="auto"
/>
`;

exports[`b renders with dir="ltr" 1`] = `
<b
  dir="ltr"
/>
`;

exports[`b renders with dir="rtl" 1`] = `
<b
  dir="rtl"
/>
`;

exports[`b renders with draggable="false" 1`] = `
<b
  draggable="false"
/>
`;

exports[`b renders with draggable="true" 1`] = `
<b
  draggable="true"
/>
`;

exports[`b renders with empty attributes 1`] = `<b />`;

exports[`b renders with hidden=false 1`] = `
<b
  hidden={false}
/>
`;

exports[`b renders with hidden=true 1`] = `
<b
  hidden={true}
/>
`;

exports[`b renders with id="TEST" 1`] = `
<b
  id="TEST"
/>
`;

exports[`b renders with inputMode="decimal" 1`] = `
<b
  inputMode="decimal"
/>
`;

exports[`b renders with inputMode="email" 1`] = `
<b
  inputMode="email"
/>
`;

exports[`b renders with inputMode="none" 1`] = `
<b
  inputMode="none"
/>
`;

exports[`b renders with inputMode="numeric" 1`] = `
<b
  inputMode="numeric"
/>
`;

exports[`b renders with inputMode="search" 1`] = `
<b
  inputMode="search"
/>
`;

exports[`b renders with inputMode="tel" 1`] = `
<b
  inputMode="tel"
/>
`;

exports[`b renders with inputMode="text" 1`] = `
<b
  inputMode="text"
/>
`;

exports[`b renders with inputMode="url" 1`] = `
<b
  inputMode="url"
/>
`;

exports[`b renders with is="foo" 1`] = `
<b
  is="foo"
/>
`;

exports[`b renders with itemID="https://google.com/" 1`] = `
<b
  itemID="https://google.com/"
/>
`;

exports[`b renders with itemProp="https://google.com/" 1`] = `
<b
  itemProp="https://google.com/"
/>
`;

exports[`b renders with itemRef="foo bar" 1`] = `
<b
  itemRef="foo bar"
/>
`;

exports[`b renders with itemScope=false 1`] = `
<b
  itemScope={false}
/>
`;

exports[`b renders with itemScope=true 1`] = `
<b
  itemScope={true}
/>
`;

exports[`b renders with itemType="https://google.com/" 1`] = `
<b
  itemType="https://google.com/"
/>
`;

exports[`b renders with lang="" 1`] = `
<b
  lang=""
/>
`;

exports[`b renders with lang="es-419" 1`] = `
<b
  lang="es-419"
/>
`;

exports[`b renders with slot="TEST" 1`] = `
<b
  slot="TEST"
/>
`;

exports[`b renders with spellCheck="false" 1`] = `
<b
  spellCheck="false"
/>
`;

exports[`b renders with spellCheck="true" 1`] = `
<b
  spellCheck="true"
/>
`;

exports[`b renders with style={"color":"#000000"} 1`] = `
<b
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`b renders with tabIndex=1 1`] = `
<b
  tabIndex={1}
/>
`;

exports[`b renders with title="TEST" 1`] = `
<b
  title="TEST"
/>
`;

exports[`b renders with translate="no" 1`] = `
<b
  translate="no"
/>
`;

exports[`b renders with translate="yes" 1`] = `
<b
  translate="yes"
/>
`;

exports[`b renders without attributes 1`] = `<b />`;
